openapi/single-sign-on/single-sign-on.yaml

```diff
@@ -0,0 +1,357 @@
+        - schema:
+            type: string
+          in: query
+          name: group_path
+          description: 'Group ID (Optional Service Context) '
+        - schema:
+            type: string
+            enum:
+              - none
+              - login
+              - consent
+              - select_account
+          in: query
+          name: prompt
+          description: Prompt controls what prompts will or will not be shown to the user. See details...
+        - schema:
+            type: string
+          in: query
+          name: redirect_uri
+          description: Redirect URI is where the User will be sent after the Authorization flow with either the result or an error.
+        - schema:
+            type: string
+            example: code
+          in: query
+          name: response_type
+          required: true
+          description: The Open ID authorization response type. Only 'code' is supported.
+      tags:
+        - Single-Sign-On
+    parameters:
+      - schema:
+          type: string
+        name: partnerID
+        in: path
+        required: true
+        description: Partner ID providing identity namespacing
+  '/{partnerID}/oauth2/logout':
+    get:
+      summary: OpenID Logout
+      operationId: openid-logout
+      responses:
+        '200':
+          description: OK
+      x-lifecycle:
+        status: trustedTester
+      description: |-
+        [Status](https://developers.vendasta.com/platform/ZG9jOjEwMTU2NTYy-versioning): `Trusted Tester`
+
+        Browsers can be directed to this endpoint to logout. This will end the User's Session at Single-Sign-On and Sessions in other services which were accessed using the Single-Sign-On Session.
+
+        This operation has no security necessarily, but logout can only act upon a browser's session via the request's `vig_session` cookie.
+      parameters:
+        - schema:
+            type: string
+          in: query
+          name: post_logout_redirect_uri
+          description: 'URI to redirect to after logout. [This must be preregistered (spec), but we do not yet require or allow this]. [spec miss: id_token_hint is required when this parameter is used.]'
+        - schema:
+            type: string
+          in: query
+          name: namespace
+          description: 'Used as the partnerID if the endpoint is called without a partnerID in the path (`/oauth2/token`), and is required when doing so.'
+        - schema:
+            type: string
+          in: cookie
+          name: vig-namespaced-session
+          description: 'Cookie for a Vendasta Identity Gateway session, stored at namespaced paths.'
+          required: true
+        - schema:
+            type: string
+          in: query
+          name: id_token_hint
+          description: '[spec miss: Identity token of user requesting logout, used to authentically indicate user and service requesting logout. Required if using `post_logout_redirect_uri`.] '
```

Comment:
The reason is really strange "security of denial attacks". 

I believe it's like: we only want to redirect to safe places (preregistered) and an id token should be required for knowing the client since they are the ones preregistering, since virtually only that service would have id tokens signed for their service (technically so do their users)". 

Further, if we do no validation on redirect we can take part in denial attacks by having traffic come from us, this oddly associated party?

So there seems to be a feature story here to allow/require this preregistering, but I don't fully grasp the motivation for us to do it. Our angular oauth2 lib already sends this param I believe, FYI.